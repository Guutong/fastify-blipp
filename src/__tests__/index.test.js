global.console.log = jest.fn();

describe("blipp", () => {
  beforeAll(() => {
    //actions = Actions(server);
  });

  beforeEach(() => {
    request = {};
    jest.clearAllMocks();
  });

  describe("/", () => {
    it("prints routes", async done => {
      const fastify = require("fastify")();

      fastify.register(require("../../src/index"));

      fastify.get("/hello/:username", async (req, reply) => ({
        greeting: `Hello, ${req.params.username}`
      }));
      fastify.get("/hello/:username/CAPS", async (req, reply) => ({
        greeting: `Hello, ${req.params.username.toUpperCase()}`
      }));
      fastify.post("/hello", async (req, reply) => ({
        greeting: `Hello, ${req.body.username}`
      }));

      await fastify.ready();

      fastify.blipp();

      expect(console.log.mock.calls[0][0]).toBe(`üè∑Ô∏è  Routes:`);
      expect(console.log.mock.calls[1][0]).toMatchSnapshot();
      done();
    });
  });
});
