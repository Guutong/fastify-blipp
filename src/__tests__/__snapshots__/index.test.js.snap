// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blipp / prints routes 1`] = `
"[32mGET[39m	/hello/[90m:username[39m
[32mPOST[39m	/hello
"
`;
